### DISCUSSION TITLE
useQuery refetch { data } lifecycle

### DISCUSSION THREAD

**Discussion by brantleyenglish** [TanStack/query#5181] [Q&A] [answered]
If a useQuery returns data on the initial load and stores it in `data`, does it ever replace that value if the next api call returned is an error? Or does it continue to show the `data` from only the last successful call?

Would running a refetch with `queryClient.invalidateQueries();` affect it any differently?

Thanks so much!

**Comment by louis-young** [comment, marked as answer]
Hi 👋 

The data that exists in the cache will still be returned by the query if it refetches erroneously. Marking a query as stale via `invalidateQueries` will exhibit the same behaviour if the query subsequently refetched erroneously. 

Hopefully this helps (also, apologies for accidentally closing the discussion; it was an accident!) 🙂
